obj_factory.h

```diff
@@ -102,6 +102,42 @@ class ObjFactory {
         return vec[idx.val].get();
     }
 
+    bool exist(const std::string& uri) {
+        if (map.find(uri) == map.end()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    bool erase(const Idx<ObjType>& idx) {
+        if (idx.val >= vec.size()) {
+            return false;
+        }
+        // erase map member
+        const auto* elem_ptr = vec[idx.val].get();
+        if (map.erase(elem_ptr->uri) == 0) {
+            return false;
+        }
+        // erase vec member
+        vec.erase(vec.begin() + idx.val);
+        return true;
```

Comment:
If you erase, you must reindex the collection.

Reply:
Which collection ?

Reply:
I forgot about this... Then a test using the object itself (and its index) would be nice: check that `obj_factory[i]->val.idx = i` (untested)

Reply:
I think it's `obj_factory[i]->idx.val = i`, this must be always true (invariant managed by obj_factory)

Reply:
Okay I have understood your remark.